### ISSUE TITLE
When using this permission, the backing application of the service principal being created must in the local tenant

### ISSUE THREAD

**Issue by jim-minter** [Azure/ARO-RP#535]
Error seen on an `az aro create` - no further context currently.

When using this permission, the backing application of the service principal being created must in the local tenant

Possibly need to catch and retry this if it happens more.

**Comment by jim-minter** [comment]
I think this must happen in `def create_service_principal(self, app_id)` - need a retry here

**Comment by jim-minter** [comment]
most of az just catches all exceptions on this call and does a sleep/retry loop here

**Comment by m1kola** [comment]
I did hit it 3 times today. Here relevant output of `az aro create` with `--debug`:

```
msrest.http_logger : Request URL: 'https://graph.windows.net/7a1815fb-63ba-4af7-8f33-e7333e421980/servicePrincipals?api-version=1.6'
msrest.http_logger : Request method: 'POST'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'Content-Length': '49'
msrest.http_logger :     'User-Agent': 'python/3.8.2 (macOS-10.15.4-x86_64-i386-64bit) msrest/0.6.13 msrest_azure/0.6.3 azure-graphrbac/0.60.0 Azure-SDK-For-Python AZURECLI/2.5.1 (HOMEBREW)'
msrest.http_logger : Request body:
msrest.http_logger : {"appId": "17eb8791-1a44-4428-b350-94861a804d18"}
msrest.universal_http : Configuring redirects: allow=True, max=30
msrest.universal_http : Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http : Configuring proxies: ''
msrest.universal_http : Evaluate proxies against ENV settings: True
urllib3.connectionpool : Starting new HTTPS connection (1): graph.windows.net:443
urllib3.connectionpool : https://graph.windows.net:443 "POST /7a1815fb-63ba-4af7-8f33-e7333e421980/servicePrincipals?api-version=1.6 HTTP/1.1" 403 284
msrest.http_logger : Response status: 403
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Content-Type': 'application/json; odata=minimalmetadata; streaming=true; charset=utf-8'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'ocp-aad-diagnostics-server-name': 'oMHfAEIp1gHYFHdJ6hb0QGsjL3oM+bEEQZMKernWW0M='
msrest.http_logger :     'request-id': '0e6defe8-65c8-48d5-abfa-990bc8a1ca6a'
msrest.http_logger :     'client-request-id': '82d1f0c0-9514-11ea-83f5-38f9d3592049'
msrest.http_logger :     'x-ms-dirapi-data-contract-version': '1.6'
msrest.http_logger :     'ocp-aad-session-key': 'REDACTED'
msrest.http_logger :     'DataServiceVersion': '3.0;'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'Access-Control-Allow-Origin': '*'
msrest.http_logger :     'X-AspNet-Version': '4.0.30319'
msrest.http_logger :     'X-Powered-By': 'ASP.NET'
msrest.http_logger :     'Duration': '2120017'
msrest.http_logger :     'Date': 'Wed, 13 May 2020 12:23:16 GMT'
msrest.http_logger :     'Content-Length': '284'
msrest.http_logger : Response content:
msrest.http_logger : {"odata.error":{"code":"Authorization_RequestDenied","message":{"lang":"en","value":"When using this permission, the backing application of the service principal being created must in the local tenant"},"requestId":"0e6defe8-65c8-48d5-abfa-990bc8a1ca6a","date":"2020-05-13T12:23:16"}}
msrest.exceptions : When using this permission, the backing application of the service principal being created must in the local tenant
cli.azure.cli.core.util : When using this permission, the backing application of the service principal being created must in the local tenant
When using this permission, the backing application of the service principal being created must in the local tenant
Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x111589ca0>]
az_command_data_logger : exit code: 1
telemetry.save : Save telemetry record of length 2629 in cache
telemetry.check : Negative: The /Users/mradchuk/.azure/telemetry.txt was modified at 2020-05-13 13:17:01.260508, which in less than 600.000000 s
command ran in 2.844 seconds.
```

**Comment by liuke0712** [comment]
@jim-minter Hi Jim, I encoutered the same issue while creating sp after creating an application, may i ask how didi you resolve it?
Much appreciate!
Maggie

**Comment by m1kola** [comment]
Hi @liuke0712,

I believe that this issue was was fixed in Azure Red Hat OpenShift command #631 by adding retries to handle AAD sync delays.

If you still experience something similar while creating an Azure Red Hat OpenShift I encourage you to reach out to Microsoft support: it will enable us to investigate this further.

Kind regards,
Mikalai
